src/applications/gqrx/mainwindow.cpp

```diff
@@ -1540,21 +1541,49 @@ void MainWindow::stopAudioStreaming()
 }
 
 /** Start I/Q recording. */
-void MainWindow::startIqRecording(const QString& recdir)
+void MainWindow::startIqRecording(const QString& recdir, const QString& format)
 {
     qDebug() << __func__;
     // generate file name using date, time, rf freq in kHz and BW in Hz
     // gqrx_iq_yyyymmdd_hhmmss_freq_bw_fc.raw
     auto freq = (qint64)(rx->get_rf_freq());
     auto sr = (qint64)(rx->get_input_rate());
     auto dec = (quint32)(rx->get_input_decim());
-    auto lastRec = QDateTime::currentDateTimeUtc().
-            toString("%1/gqrx_yyyyMMdd_hhmmss_%2_%3_fc.'raw'")
-            .arg(recdir).arg(freq).arg(sr/dec);
+    auto currentDate = QDateTime::currentDateTimeUtc();
+    auto filenameTemplate = currentDate.toString("%1/gqrx_yyyyMMdd_hhmmss_%2_%3_fc.%4").arg(recdir).arg(freq).arg(sr/dec);
+    bool sigmf = (format == "SigMF");
+    auto lastRec = filenameTemplate.arg(sigmf ? "sigmf-data" : "raw");
+
+    QFile metaFile(filenameTemplate.arg("sigmf-meta"));
+    bool ok = true;
+    if (sigmf) {
+        auto meta = QJsonDocument { QJsonObject {
+            {"global", QJsonObject {
+                {"core:datatype", "cf32_le"}, // does anyone use gqrx on big endian platforms?
```

Comment:
Running on big-endian systems is rare but not unheard-of. We could use this:

https://doc.qt.io/qt-6.2/qtglobal.html#Q_BIG_ENDIAN

```suggestion
#if Q_BYTE_ORDER == Q_BIG_ENDIAN
                {"core:datatype", "cf32_be"},
#else
                {"core:datatype", "cf32_le"},
#endif
```